tests/test_mongita.py

```diff
@@ -634,6 +635,24 @@ def test_update_many(client_class):
     assert set(d.get('weight') for d in coll.find({})) == \
            set(d['weight'] + 1 if isinstance(d.get('weight'), Number) and d.get('weight') < 4 else d.get('weight') for d in TEST_DOCS)
 
+@pytest.mark.parametrize("client_class", CLIENTS)
+def test_push_update(client_class):
+    client, coll, imr = setup_many(client_class)
+
+    array_field = "continents"
+    push_value = "foobar"
+    array_initial = copy.copy(coll.find_one({})[array_field])
+
+    coll.update_one({}, {"$push": {array_field: push_value}})
+
+    array_final = coll.find_one({})[array_field]
+
+    print(f'  initial\n{array_initial}\n')
+    print(f'  final\n{array_final}')
+    assert len(array_final) == len(array_initial) + 1
+    assert array_final == [*array_initial, push_value] 
```

Comment:
The tests pass but one issue I see is that the ClientDisk previous test runs seems to be leaking into the current test runs.

Example

mongita.mongita_client.MongitaClientMemory
```
# M
  initial
['AF']

  final
['AF', 'foobar']
```

mongita.mongita_client.MongitaClientDisk
```
  initial
['AF', 'foobar']

  final
['AF', 'foobar', 'foobar']
```

Reply:
I took a first pass. The leaking is an interesting problem. I'm pretty sure it's because child lists are not copied in MongitaClientMemory and therefore the list in memory is the same as the list that was added. Sure enough, if you do a copy.deepcopy on document insert/replace, the problem you've observed goes away. I'm concerned that there are unreported bugs because of this that I'll need to search out and explicitly test for.

Otherwise, I think the only thing that needs to be added is error checking for non-lists at that key and creation logic when the key is not present according to https://docs.mongodb.com/manual/reference/operator/update/push/. To be fair, I think some error checking also needs to be added to $inc.

Will think more about this and try to address in the next day or two.